tests/unit_tests/utils/test_model_downloader.py

```diff
@@ -0,0 +1,62 @@
+from pathlib import Path
+from typing import Union
+from unittest.mock import create_autospec, MagicMock, call
+
+from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
+
+from exasol_transformers_extension.utils.model_downloader import ModelDownloader, ModelFactoryProtocol
+from exasol_transformers_extension.utils.temporary_directory_factory import TemporaryDirectoryFactory
+from exasol_transformers_extension.utils.bucketfs_model_uploader import BucketFSModelUploader, \
+    BucketFSModelUploaderFactory
+from tests.utils.mock_cast import mock_cast
+
+
+class TestSetup:
+    def __init__(self):
+        self.bucketfs_location: Union[BucketFSLocation, MagicMock] = create_autospec(BucketFSLocation)
+        self.model_factory: Union[ModelFactoryProtocol, MagicMock] = create_autospec(ModelFactoryProtocol)
+        self.temporary_directory_factory: Union[TemporaryDirectoryFactory, MagicMock] = \
+            create_autospec(TemporaryDirectoryFactory)
+        self.bucketfs_model_uploader_factory: Union[BucketFSModelUploaderFactory, MagicMock] = \
+            create_autospec(BucketFSModelUploaderFactory)
+        self.bucketfs_model_uploader: Union[BucketFSModelUploader, MagicMock] = \
+            create_autospec(BucketFSModelUploader)
+        mock_cast(self.bucketfs_model_uploader_factory.create).side_effect = [self.bucketfs_model_uploader]
+
+        self.token = "token"
+        self.model_name = "test_model_name"
+        self.model_path = Path("test_model_path")
+        self.downloader = ModelDownloader(
+            bucketfs_location=self.bucketfs_location,
+            model_path=self.model_path,
+            model_name=self.model_name,
+            token=self.token,
+            temporary_directory_factory=self.temporary_directory_factory,
+            bucketfs_model_uploader_factory=self.bucketfs_model_uploader_factory
+        )
+
+
+def test_init():
+    test_setup = TestSetup()
+    assert test_setup.temporary_directory_factory.mock_calls == [] \
+           and test_setup.model_factory.mock_calls == [] \
+           and test_setup.bucketfs_location.mock_calls == [] \
```

Comment:
i am assuming this checks if the mock is actually new and actually exists?

Reply:
no this checks if the mock was called by the testee, in this specific case, the test setup calls the constructor of the testee, such that we only instantiate the test setup and test afterward, if the mocks were called,